### PR TITLE
Toxin buildup

### PR DESCRIPTION
**Pull request by EvilMelonTheSecond**
## About The Pull Request
An alternative to #8772. Light neurotox now gains purge with the higher effect tiers.

## Why It's Good For The Game
Instead of being almost completely ignorable light neurotoxin now builds up to being the same as neurotoxin if not purged from the metabolism. Range xeno are rewarded for focusing down marines with their chem combat and smart marines can avoid getting purged from range, the peak of balance.

Numbers are completely arbitrary, should be adjusted if necessary.
The current numbers are:
0u purge for first stage
0.2u purge for the second stage
1u purge for third/final stage

## Changelog
:cl:
balance: Light neurotox now has purge if metabolized too long
/:cl:

### CODE REVIEW

**Comment by EvilMelonTheSecond** [comment]
Please ignore the wall of commits, they are there to keep me company at night.

File: code/modules/reagents/reagents/toxin.dm [hunk 1]
```diff
@@ -486,10 +485,12 @@
 			power = (2*effect_str) //While stamina loss is going, stamina regen apparently doesn't happen, so I can keep this smaller.
 			L.reagent_pain_modifier -= PAIN_REDUCTION_LIGHT
 		if(21 to 45)
+			purge_rate = (purge_rate == 1) ? 1 : 0.2
 			power = (6*effect_str)
 			L.reagent_pain_modifier -= PAIN_REDUCTION_HEAVY
 			L.jitter(4) //Shows that things are bad
 		if(46 to INFINITY)
+			purge_rate = (purge_rate == 1) ? 1 : 1
```
Comment (BraveMole) [inline code/modules/reagents/reagents/toxin.dm:493]: just directly set the purge rate as 1 here, and 0.2 at l 488.
These check are useless

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by BraveMole — CHANGES_REQUESTED** [review]
@David-Stormwell

File: code/modules/reagents/reagents/toxin.dm [hunk 2]
```diff
@@ -490,6 +489,7 @@
 			L.reagent_pain_modifier -= PAIN_REDUCTION_HEAVY
 			L.jitter(4) //Shows that things are bad
 		if(46 to INFINITY)
+			purge_rate = 1 //Even neurotoxins with no purge will start purging in the last stage
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/reagents/reagents/toxin.dm:492]: why are you even affecting normal neurotox

[the remark this answers is not in the document]
Comment (EvilMelonTheSecond) [reply code/modules/reagents/reagents/toxin.dm:492]: The reason is I'm bad at code. I dunno how to call parent function and modify it afterwards

**Review by BraveMole — APPROVED** [review]

**Review by BraveMole — APPROVED** [review]

**Pull request merged**